### PR TITLE
migrate to poetry for package management

### PR DESCRIPTION
**Pull request by HenryFBP**
### Background

Seemed relatively easy to do, so I migrated `requirements.txt` to Poetry.

### Changes

1. Migrate package manager to Poetry
2. Update documentation

### Test Plan

I tested the local install by using Poetry, and Docker with `docker build . -t test; docker run -it test`.

### CODE REVIEW

**Comment by LuisLechugaRuiz** [comment]
This would be great to have! I think to start scaling the repo we should first make it installable using Pip so we can combine with other features

**Comment by dominikdassow** [comment]
> This would be great to have! I think to start scaling the repo we should first make it installable using Pip so we can combine with other features

Agree!

**Comment by nponeccop** [comment]
@HenryFBP There are conflicts now

File: README.md
Comment (rickythefox) [inline README.md]: Don't think you need `poetry shell` when using `poetry run`.

File: Dockerfile
Comment (rickythefox) [inline Dockerfile:10]: You could just use one COPY statement and save some layers.

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@HenryFBP There are conflicts again

File: .env.template
Comment (nponeccop) [inline .env.template:5]: These changes creeped in during auto-merge

File: .env.template
Comment (nponeccop) [inline .env.template:12]: And these ones

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (nponeccop) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: And these ones

File: Dockerfile
Comment (meltingscales) [reply Dockerfile:10]: Thanks, good idea. I added the 2 files to 1 COPY statement.

Comment (rickythefox) [reply Dockerfile:10]: Can't you also put `scripts/` into the same statement?

**Comment by nponeccop** [comment]
@HenryFBP No, you should roll them back by yourself, there are several ways, sometimes e.g. checkout -p is useful.

**Comment by HenryFBP** [comment]
Ah, no worries.

I'm not actually sure how to do that. 

Maybe it would just be easier to make a new branch from master and apply my changes again, and make a new PR.

**Comment by nponeccop** [comment]
Just remember that due to many merges you'll have to merge conflicts over and over again before it finally merges. So learning how to redo a botched PR without removing it is a must. It's all too bad, and it's up to you, but we are facing a surge of PRs and many people fix their PRs without redoing. Sometimes redoing is the simplest way ofc.

**Comment by HenryFBP** [comment]
> Just remember that due to many merges you'll have to merge conflicts over and over again before it finally merges. So learning how to redo a botched PR without removing it is a must. It's all too bad, and it's up to you, but we are facing a surge of PRs and many people fix their PRs without redoing. Sometimes redoing is the simplest way ofc.

Sounds kind of complicated, surely there's a better way? I mean, I'm more inclined to just redo it, I'm not sure why I'd want to learn how to redo a botched PR if it's just faster and easier to re-do the PR.

**Comment by rickythefox** [comment]
> > Just remember that due to many merges you'll have to merge conflicts over and over again before it finally merges. So learning how to redo a botched PR without removing it is a must. It's all too bad, and it's up to you, but we are facing a surge of PRs and many people fix their PRs without redoing. Sometimes redoing is the simplest way ofc.
> 
> Sounds kind of complicated, surely there's a better way? I mean, I'm more inclined to just make a new one, I'm not sure why I'd want to learn how to redo a botched PR if it's just faster and easier to make a new PR.

I would stash the files you changed (or just copy them somewhere). Then:
```git remote add upstream https://github.com/Torantulino/Auto-GPT/
git fetch upstream
git checkout master
git reset --hard upstream/master  
```
Your fork is now equal to upstream. Now unstash or copy the changed files, commit and force-push
``` git push origin master --force
```

No need to open a new PR.

**Comment by HenryFBP** [comment]
@rickythefox Thank you! I did that, and it worked.

**Comment by nponeccop** [comment]
Apparently it didn't. Maybe someone can volunteer to help? The number of files changes should be 2 or 3, not 55.

**Comment by rickythefox** [comment]
I can help but would need write access to the fork at https://github.com/HenryFBP/Auto-GPT . Alternatively I can submit a new PR but don't want to steal the fame here. =)

**Comment by nponeccop** [comment]
@HenryFBP do you mind that @rickythefox makes a mergeable PR out of your work?

**Comment by rickythefox** [comment]
@HenryFBP Or add me as collaborator on your fork

**Comment by nponeccop** [comment]
Even I am not a collaborator. It's a long story, it's a personal repo and there are no permissions, so only @Torantulino , @p-i- and @richbeales have push rights.

**Comment by rickythefox** [comment]
@nponeccop on the fork, not on the repo =)

**Comment by rickythefox** [comment]
@nponeccop submitted a PR so you can merge if @HenryFBP is fine with that approach.

**Comment by HenryFBP** [comment]
> @nponeccop submitted a PR so you can merge if @HenryFBP is fine with that approach.

Sorry for the late reply, been AFK. Sure, I'm fine with anything that gets the code into the main branch.

**Comment by nponeccop** [comment]
What is the PR number? Is it merged yet? Does it have conflicts? Does it have an approval from me?

**Comment by rickythefox** [comment]
@nponeccop saw that you saw and approved it. 🙏

**Comment by nponeccop** [comment]
Don't forget to close this one once that one is merged. Likely today.

**Comment by p-i-** [comment]
I can't merge this. It says 52 files changed. All sorts of changes unrelated  to poetry.

I like Poetry. It does a great job of being able to tree-view the dep-tree and auto-manage version-conflicts.

We need a fresh PR I think. This has gotten messy.

Tentatively I'm gona close this one. Apologies for slow team response to PRs ... we're getting there!

**Comment by nponeccop** [comment]
@p-i- This is an outdated version of #1102. Check there.

**Pull request closed without merging**
